packages/core/src/app/shipping/isSelectedShippingOptionValid.ts

```diff
@@ -0,0 +1,17 @@
+import { Consignment } from '@bigcommerce/checkout-sdk';
+import { every } from 'lodash';
+
+export default function isSelectedShippingOptionValid(consignments: Consignment[]): boolean {
+    if (!consignments.length) {
+        return false;
+    }
+
+    return every(
+        consignments,
+        (consignment) =>
+        (consignment.availableShippingOptions &&
+            consignment.availableShippingOptions.filter(
+                ({ id }) => id === consignment.selectedShippingOption?.id,
+            ).length)
```

Comment:
If selectedShippingOption is null, maybe don't need to loop the filter?
```suggestion
        (consignment.availableShippingOptions &&
         consignment.selectedShippingOption &&
            consignment.availableShippingOptions.filter(
                ({ id }) => id === consignment.selectedShippingOption.id,
            ).length)
```

Reply:
It will not come to this function if there is no `selectedShippingOption`, since that is already checked via `hasSelectedShippingOptions`